### ISSUE TITLE
Passing new value(s) to `selected` doesn't correctly filter the menu

### ISSUE THREAD

**Issue by tko** [ericgio/react-bootstrap-typeahead#198]
### Version
1.4.0

### Steps to reproduce
0. have two autocomplete entries linked with each other such that selecting an option in one updates the other to match. For example adding/selecting name+email pairs.
1. select value from one entry
2. click open both of the dropdowns
3. select a different value from second entry
4. click open both of the dropdowns

### Expected Behavior
Both dropdowns / entries show similar options, shouldn't matter whether the value was selected through clicking on the entry itself or updated programmatically.

### Actual Behavior
After selecting `one` on the first entry (step 1&2):
<img width="210" alt="first-one" src="https://user-images.githubusercontent.com/13787/27083273-4f29c43c-5048-11e7-8c0b-b111a153689f.png"><img width="210" alt="second-one" src="https://user-images.githubusercontent.com/13787/27083276-4f56fe98-5048-11e7-9c7c-0ee57d9863e6.png">

After selecting `two` from the second entry (3&4):
<img width="209" alt="first-two" src="https://user-images.githubusercontent.com/13787/27083274-4f397a08-5048-11e7-9ca3-10b14f79d3e9.png"><img width="211" alt="second-two" src="https://user-images.githubusercontent.com/13787/27083275-4f50953a-5048-11e7-9e63-14cac735c99d.png">

### Code
```jsx
<Typeahead
    onInputChange={(selected) => this.setState({ selected })}
    options={['one', 'two', 'three']}
    selected={[this.state.selected]}
/>
<Typeahead
    onInputChange={(selected) => this.setState({ selected })}
    options={['one', 'two', 'three']}
    selected={[this.state.selected]}
/>
```

**Comment by ericgio** [comment]
As noted in your other issue, use `onChange` not `onInputChange`.

**Comment by tko** [comment]
Right, my use of `onInputChange` was unintentional here, _however_ considering I'm ignoring `onChange` completely and _only_ interacting with the dropdowns I believe there is no difference in behavior. I think I rechecked this with `onChange` and using `{name, email}` objects instead of strings but as it was behaving the same thought the reduced test case was more useful.

**Comment by ericgio** [comment]
I see the issue here.

**Comment by ericgio** [comment]
The issue here is that `componentWillReceiveProps` is updating `selected` but doesn't update `text` as well.

**Comment by ericgio** [comment]
Fixed in v2.0.0-alpha.1
